prism_compile.c

```diff
@@ -751,18 +751,33 @@ pm_interpolated_node_compile(pm_node_list_t *parts, rb_iseq_t *iseq, NODE dummy_
                     current_string = string_value;
                 }
             }
-            else {
+            else if (PM_NODE_TYPE_P(part, PM_EMBEDDED_STATEMENTS_NODE) &&
+                    ((pm_embedded_statements_node_t *)part)->statements->body.size == 1 &&
+                    PM_NODE_TYPE_P(((pm_embedded_statements_node_t *)part)->statements->body.nodes[0], PM_STRING_NODE)) {
+                pm_string_node_t *string_node = (pm_string_node_t *)((pm_embedded_statements_node_t *)part)->statements->body.nodes[0];
+                VALUE string_value = parse_string_encoded((pm_node_t *)string_node, &string_node->unescaped, parser);
                 if (RTEST(current_string)) {
-                    current_string = rb_fstring(current_string);
-                    if (parser->frozen_string_literal) {
-                        ADD_INSN1(ret, &dummy_line_node, putobject, current_string);
-                    }
-                    else {
-                        ADD_INSN1(ret, &dummy_line_node, putstring, current_string);
-                    }
-                    current_string = Qnil;
-                    number_of_items_pushed++;
+                    current_string = rb_str_concat(current_string, string_value);
+                }
+                else {
+                    current_string = string_value;
+                }
+            }
+            else {
+                if (!RTEST(current_string)) {
+                    rb_encoding *enc = rb_enc_from_index(rb_enc_find_index(parser->encoding->name));
+                    current_string = rb_enc_str_new(NULL, 0, enc);
                 }
+
+                if (parser->frozen_string_literal) {
+                    ADD_INSN1(ret, &dummy_line_node, putobject, rb_str_freeze(current_string));
```

Comment:
Should this be an fstring?

Reply:
No, I don't think so. There are individual parts of the string rather than the resulting string. There is a `rb_fstring` call later on when we finish building the string.